modules/dxp/apps/scim/scim-configuration-web/src/main/java/com/liferay/scim/configuration/web/internal/portlet/action/SaveScimConfigurationMVCActionCommand.java

```diff
@@ -155,6 +169,48 @@ else if (Objects.equals(cmd, "reset")) {
 
 			_oAuth2ApplicationLocalService.deleteOAuth2Application(
 				oAuth2Application);
+
+			Indexer<User> userIndexer = IndexerRegistryUtil.nullSafeGetIndexer(
+				User.class);
+
+			Indexer<UserGroup> userGroupIndexer =
+				IndexerRegistryUtil.nullSafeGetIndexer(UserGroup.class);
+
+			ActionableDynamicQuery query =
+				_expandoValueLocalService.getActionableDynamicQuery();
+
+			query.setAddCriteriaMethod(
+				dynamicQuery -> {
+					Property dataProperty = PropertyFactoryUtil.forName("data");
+
+					dynamicQuery.add(dataProperty.eq(scimClientId));
+				});
+
+			query.setPerformActionMethod(
+				(ExpandoValue expandoValue) -> {
+					_expandoRowLocalService.deleteRow(
+						expandoValue.getTableId(), expandoValue.getClassPK());
+
+					ClassName className = _classNameLocalService.getClassName(
+						expandoValue.getClassNameId());
+
+					String classNameString = className.getClassName();
+
+					if (classNameString.equals(User.class.getName())) {
+						userIndexer.reindex(
+							_userLocalService.getUser(
+								expandoValue.getClassPK()));
+					}
+					else if (classNameString.equals(
```

Comment:
Is there every a situation in SCIM where the `classNameString` won't equal `User.class.getName()` or `UserGroup.class.getName()`?  If those are the only two possible options, then the `else if()` should just be `else`.

Reply:
Only user and userGroup are used as for now, i just left the else if to be precise/clear. but i can change it if needed

Reply:
That does make sense, however it also adds an unnecessary String comparison, and the next line tells the user exactly what is happening:

> userGroupIndexer.reindex(
> 	_userGroupLocalService.getUserGroup(
> 		expandoValue.getClassPK()));

I'll adjust this in my commits